@yahya-darman Would you mind fixing these NNLS test failures?

```
FAILED scipy/optimize/tests/test_nnls.py::TestNNLS::test_nnls_tall - DeprecationWarning: Arguments {'atol'} are deprecated, whether passed by position or keyword. They will be removed in SciPy 1.18.0. The 'atol' parameter is deprecated and will be removed in SciPy 1.18.0. It is not used in the implementation.
FAILED scipy/optimize/tests/test_nnls.py::TestNNLS::test_nnls_wide - DeprecationWarning: Arguments {'atol'} are deprecated, whether passed by position or keyword. They will be removed in SciPy 1.18.0. The 'atol' parameter is deprecated and will be removed in SciPy 1.18.0. It is not used in the implementation.
FAILED scipy/optimize/tests/test_nnls.py::TestNNLS::test_nnls_inner_loop_case2 - DeprecationWarning: Arguments {'atol'} are deprecated, whether passed by position or keyword. They will be removed in SciPy 1.18.0. The 'atol' parameter is deprecated and will be removed in SciPy 1.18.0. It is not used in the implementation.
```

It looks like there are parts of the NNLS tests that use `atol`. The simplest thing would be to remove `atol` there. (Note: don't remove atol from `assert_allclose` - that still does something!)

Would you mind also fixing this lint error? https://github.com/scipy/scipy/actions/runs/12921459761/job/36035528638?pr=22386

You can check locally if the tests have been fixed using the command

```
python dev.py test -s optimize
```

and you can check the lint locally using the command

```
python dev.py lint
```

--- a/scipy/optimize/_nnls.py
+++ b/scipy/optimize/_nnls.py
@@ -12,7 +12,6 @@ _nnls_dep_message = ("The 'atol' parameter is deprecated and will be removed in 
                             deprecated_args={'atol'},
                             custom_message=_nnls_dep_message)
 def nnls(A, b, *, maxiter=None, atol=_NoValue):
-    """
     Solve ``argmin_x || Ax - b ||_2`` for ``x>=0``.
 
     This problem, often called as NonNegative Least Squares, is a convex

--- a/scipy/optimize/_nnls.py
+++ b/scipy/optimize/_nnls.py
@@ -12,6 +12,7 @@ _nnls_dep_message = ("The 'atol' parameter is deprecated and will be removed in 
                             deprecated_args={'atol'},
                             custom_message=_nnls_dep_message)
 def nnls(A, b, *, maxiter=None, atol=_NoValue):
+    """
     Solve ``argmin_x || Ax - b ||_2`` for ``x>=0``.
 
     This problem, often called as NonNegative Least Squares, is a convex

--- a/scipy/optimize/_nnls.py
+++ b/scipy/optimize/_nnls.py
@@ -9,10 +9,9 @@ _nnls_dep_message = ("The 'atol' parameter is deprecated and will be removed in 
                      "SciPy 1.18.0. It is not used in the implementation.")
 
 @_deprecate_positional_args(version='1.18.0',
-                           deprecated_args={'atol'},
-                           custom_message=_nnls_dep_message)
-
-def nnls(A, b, maxiter=None, *, atol=_NoValue):
+                            deprecated_args={'atol'},
+                            custom_message=_nnls_dep_message)
+def nnls(A, b, *, maxiter=None, atol=_NoValue):
     """
     Solve ``argmin_x || Ax - b ||_2`` for ``x>=0``.
 

--- a/scipy/optimize/tests/test_nnls.py
+++ b/scipy/optimize/tests/test_nnls.py
@@ -21,7 +21,7 @@ class TestNNLS:
         x = np.abs(self.rng.uniform(low=-2, high=2, size=[10]))
         x[::2] = 0
         b = a @ x
-        xact, rnorm = nnls(a, b, atol=500*np.linalg.norm(a, 1)*np.spacing(1.))
+        xact, rnorm = nnls(a, b)
         assert_allclose(xact, x, rtol=0., atol=1e-10)
         assert rnorm < 1e-12
 
@@ -32,7 +32,7 @@ class TestNNLS:
         x = np.abs(self.rng.uniform(low=-2, high=2, size=[120]))
         x[::2] = 0
         b = a @ x
-        xact, rnorm = nnls(a, b, atol=500*np.linalg.norm(a, 1)*np.spacing(1.))
+        xact, rnorm = nnls(a, b)
         assert_allclose(xact, x, rtol=0., atol=1e-10)
         assert rnorm < 1e-12
 
@@ -153,7 +153,7 @@ class TestNNLS:
         k[nz] = 0
         W = np.diag(w)
 
-        dact, _ = nnls(W @ A, W @ k, atol=1e-7)
+        dact, _ = nnls(W @ A, W @ k)
 
         p = np.cumsum(dact)
         assert np.all(dact >= 0)
@@ -433,6 +433,7 @@ class TestNNLS:
         a = np.array([[1, 0], [1, 0], [0, 1]])
         b = np.array([2, 1, 1])
         
-        message = "The 'atol' parameter is deprecated and will be removed in SciPy 1.18.0"
+        message = ("The 'atol' parameter is deprecated and will be removed in "
+                  "SciPy 1.18.0")
         with pytest.warns(DeprecationWarning, match=message):
             nnls(a, b, atol=1e-8)

